framework/include/materials/ParsedMaterialHelper.h

```diff
@@ -76,6 +76,23 @@ class ParsedMaterialHelper : public FunctionMaterialBase<is_ad>, public Function
                      const std::vector<PostprocessorName> & postprocessor_names,
                      const std::vector<std::string> & tol_names,
                      const std::vector<Real> & tol_values);
+  void functionParse(const std::string & function_expression,
+                     const std::vector<std::string> & constant_names,
+                     const std::vector<std::string> & constant_expressions,
+                     const std::vector<std::string> & mat_prop_names,
+                     const std::vector<std::string> & tol_names,
+                     const std::vector<Real> & tol_values,
+                     const std::vector<MooseFunctorName> & functor_names,
+                     const std::vector<std::string> & functor_symbols);
+  void functionParse(const std::string & function_expression,
```

Comment:
doxygen to help discern these long function signatures would be great

Reply:
I'm not quite sure what to do here. Can you please provide more details?

Reply:
duplicate the docstring
```
/**
  * Does this
   * @param function_expression functional expression to parse
   * ...
 ```